webserver/src/main/java/io/kestra/webserver/services/ai/gemini/GeminiAiService.java

```diff
@@ -0,0 +1,362 @@
+                    - IMPORTANT: Beside of the two above scenario which don't generate YAML but an error message, output only the raw YAML, nothing else.
+                    
+                    IMPORTANT: Strictly adhere to 'flowSchema'. Do not hallucinate or invent properties or types.""".formatted(NON_FLOW_REQUEST_ERROR)
+            )
+            .build();
+    }
+
+    private static LoopAgent createYamlFixingAgent(Gemini model, TenantService tenantService, FlowService flowService) {
+        LlmAgent yamlFixerAgent = LlmAgent.builder()
+            .name("kestra_flow_yaml_fixer")
+            .description("An agent to build a Kestra Flow YAML.")
+            .beforeAgentCallback((beforeAgentCallback) -> {
+                String responseForUser = (String) beforeAgentCallback.state().get("responseForUser");
+                if (responseForUser == null || responseForUser.isBlank()) {
+                    return shortCircuitLlmResponse(beforeAgentCallback, UNABLE_TO_GENERATE_FLOW_ERROR);
+                }
+
+                if (POSSIBLE_ERROR_MESSAGES.contains(responseForUser)) {
+                    return shortCircuitLlmResponse(beforeAgentCallback, responseForUser);
+                }
+
+                if (responseForUser.startsWith("```")) {
+                    responseForUser = responseForUser.replaceAll("\\s?```(?:yaml)?\\s?", "");
+                }
+
+                beforeAgentCallback.state().put("responseForUser", responseForUser);
+
+                List<ValidateConstraintViolation> validate = flowService.validate(tenantService.resolveTenant(), responseForUser);
+
+                List<String> constraints = validate.stream().map(ValidateConstraintViolation::getConstraints).filter(c -> c != null && !c.isEmpty()).toList();
+                beforeAgentCallback.state().put("validationErrors", String.join("\n", constraints));
+
+                if (constraints.isEmpty()) {
+                    String currentFlowYamlInstruction = beforeAgentCallback.userContent().get().parts().get().getFirst().text().get();
+                    String initialYaml = currentFlowYamlInstruction.substring("Current Flow YAML:\n```yaml\n".length(), currentFlowYamlInstruction.lastIndexOf("\n```"));
+                    if (initialYaml.equals(responseForUser)) {
+                        responseForUser = ALREADY_VALID_FLOW;
+                        beforeAgentCallback.state().put("responseForUser", responseForUser);
+                    }
+                    return shortCircuitLlmResponse(beforeAgentCallback, responseForUser);
+                }
+
+                return Maybe.empty();
+            })
+            .model(model)
+            .instruction(
+                """
+                    You are part of a Kestra Flow builder service. You are an expert in debugging incorrect Kestra Flow YAML. You should fix the provided validation issues and ONLY THOSE: `{{validationErrors}}`
```

Comment:
```suggestion
                    You are an expert in debugging incorrect Kestra Flow YAML. Fix only the provided validation issues: `{{validationErrors}}`
```
again, the part of flow builder service should have no effect